src/main/java/Refactoring.java

```diff
@@ -1,7 +1,41 @@
+import java.util.ArrayList;
+import java.util.List;
+
 import org.eclipse.jdt.core.dom.ASTNode;
+import org.eclipse.jdt.core.dom.Expression;
+import org.eclipse.jdt.core.dom.InfixExpression;
+import org.eclipse.jdt.core.dom.ParenthesizedExpression;
+import org.eclipse.jdt.core.dom.PrefixExpression;
 import org.eclipse.jdt.core.dom.rewrite.ASTRewrite;
 
 public abstract class Refactoring {
 	public abstract boolean isApplicable(ASTNode node);
+
 	public abstract void apply(ASTNode node, ASTRewrite rewriter);
+
+	/**
+	 * Recursively analyze an expression to get all the expressions that comprise it
```

Comment:
This looks identical to the similar code in another PR that I've already reviewed, so I'm ignoring it for now. Make sure to merge in your changes from that PR.

In the future, it's better to PR shared logic like this separately, to avoid merge conflicts.